### PR TITLE
[IMPROVE] Rewrite Two Factor Authentication Modal to React

### PR DESCRIPTION
**Pull request by gabriellsh**
## Steps to test or reproduce

- Go to Admin > Accounts > Two Factor Authentication and enable the following settings:
  - Enable Two Factor Authentication
  - Enable Two Factor Authentication via TOTP
  - Enable Two Factor Authentication via Email
  - Enforce Password fallback

- Now go to My Account > Security > Two Factor Authentication
- Click to enable Two factor via TOTP or Email (I use Authy on iOS for TOTP and configured a local email server to be able to receive the code for email)
- Once 2FA is enabled, try to disable it. The new modal for 2FA for the enabled method should pop up. You can also try to logout and login again, and the modal should be displayed.

- To test the password fallback, try to change and admin setting. It should require your password in order to save. (Beware that once you successfully type your password in once, it won't ask for it again for some time)

### CODE REVIEW

File: client/components/TwoFactorModal/TwoFactorPassword.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { useTranslation } from '../../contexts/TranslationContext';
+import GenericModal from '../GenericModal';
+import { Method, OnConfirm } from './TwoFactorModal';
+
+type TwoFactorPasswordProps = {
+	onConfirm: OnConfirm;
+	onClose: () => void;
+};
+
+const TwoFactorPassword = ({ onConfirm, onClose }: TwoFactorPasswordProps): ReactElement => {
```
Comment (dougfabris) [inline client/components/TwoFactorModal/TwoFactorPassword.tsx:14]: I think the component name should be more descriptive like "TwoFactorPasswordModal"

File: client/components/TwoFactorModal/TwoFactorTotp.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import { useTranslation } from '../../contexts/TranslationContext';
+import GenericModal from '../GenericModal';
+import { Method, OnConfirm } from './TwoFactorModal';
+
+type TwoFactorTotpProps = {
+	onConfirm: OnConfirm;
+	onClose: () => void;
+};
+
+const TwoFactorTotp = ({ onConfirm, onClose }: TwoFactorTotpProps): ReactElement => {
```
Comment (dougfabris) [inline client/components/TwoFactorModal/TwoFactorTotp.tsx:14]: I think the component name should be more descriptive like "TwoFactorTotpModal"

File: client/components/TwoFactorModal/TwoFactorEmail.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import GenericModal from '../GenericModal';
+import { Method, OnConfirm } from './TwoFactorModal';
+
+type TwoFactorEmailProps = {
+	onConfirm: OnConfirm;
+	onClose: () => void;
+	emailOrUsername: string;
+};
+
+const TwoFactorEmail = ({
```
Comment (dougfabris) [inline client/components/TwoFactorModal/TwoFactorEmail.tsx:17]: I think the component name should be more descriptive like "TwoFactorEmailModal"

File: client/components/TwoFactorModal/TwoFactorTotp.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	};
+
+	const onChange = ({ currentTarget }: ChangeEvent<HTMLInputElement>): void => {
+		setCode(currentTarget.value);
+	};
+
+	return (
+		<GenericModal
+			onConfirm={onConfirmTotpCode}
+			onCancel={onClose}
```
Comment (dougfabris) [inline client/components/TwoFactorModal/TwoFactorTotp.tsx:30]: `onCancel` is dispatching unnecessary error

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: client/components/TwoFactorModal/TwoFactorTotp.tsx (hunk 4 above)
Comment (gabriellsh) [reply client/components/TwoFactorModal/TwoFactorTotp.tsx:30]: As we talked, the error is not unecessary, as it depicts the action wasn't succesfull if TwoFactor is canceled.

**Review by dougfabris — APPROVED** [review]

**Pull request merged**